### ISSUE TITLE
Regarding Global Event Manager in RG 3.0

### ISSUE THREAD

**Issue by viju85** [roboguice/roboguice#288]
I am having trouble with global event manager in RG 3.0

I am using RoboGuice 3.0.1 along with RoboBlender in my Android application. I wanted to have a global event manager and noticed that RG3.0 already has it as mentioned here:https://github.com/roboguice/roboguice/issues/150 

I have added

bind(EventManager.class).annotatedWith(Names.named(DefaultRoboModule.GLOBAL_EVENT_MANAGER_NAME)).to(EventManager.class).asEagerSingleton();
to my custom module class inside configure().

I have a singleton WebUtil.class that performs a web API request. I want to make use of the global event manager to send out the event once the API request is complete. I have injected the eventManager using:

@Inject EventManager eventManager;
Events are sent out using:

eventManager.fire(new MyAPIEvent());
I have an Observer method in my Activity class:

handleAPICallback(@Observes MyAPIEvent apiEvent) {
// do something with apiEvent object
}
But I don't see the handleAPICallback() being called. I searched online and could not find a single example for the GlobalEventManager.

Any help would be greatly appreciated.

Stackoverflow link: http://stackoverflow.com/questions/27616198/global-event-manager-roboguice

**Comment by stephanenicolas** [comment]
I agree that the wiki could be more clear for people not so familiar with Events in RG. 

To use the global event manager, which means you want to communicate from a context to a different context, simply use : 

```java
//get the global event manager
@Named(DefaultRoboModule.GLOBAL_EVENT_MANAGER_NAME) 
@Inject EventManager globalEventManager;
....
//somewhere in code
globalEventManager.fire(new MyAPIEvent());
```

Do not register it in your module, this is already done in DefaultRoboModule. 

To receive the events, 
```java
//get the global event manager
@Named(DefaultRoboModule.GLOBAL_EVENT_MANAGER_NAME) 
@Inject EventManager globalEventManager;

globalEventManager.unregisterObserver(eventClass, listener);


handleAPICallback(@Observes MyAPIEvent apiEvent) {
// do something with apiEvent object
}
```

**Comment by cherrydev** [comment]
What made this particularly troublesome for me is that I had some components that WERE having their `@Observes` registered with the global event manager, but it only worked if they were configured as eager singletons.
